x-pack/plugins/secops/public/components/load_more_table/index.tsx

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiBasicTable,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiTitle,
+} from '@elastic/eui';
+import { isEmpty } from 'lodash/fp';
+import React from 'react';
+import styled from 'styled-components';
+
+import { EventItem } from '../../../common/graphql/types';
+import { LoadingPanel } from '../loading';
+
+export interface HoryzontalBarChartData {
+  x: number;
+  y: string;
+}
+
+export interface ItemsPerRow {
+  text: string;
+  numberOfRow: number;
+}
+
+interface BasicTableProps {
+  // tslint:disable-next-line:no-any
+  pageOfItems: any[];
+  columns: Columns[];
+  title: string | React.ReactNode;
+  loading: boolean;
+  loadingTitle?: string;
+  hasNextPage: boolean;
+  loadMore: () => void;
+  updateLimitPagination: (limit: number) => void;
+  itemsPerRow?: ItemsPerRow[];
+  limit: number;
+}
+
+interface BasicTableState {
+  isPopoverOpen: boolean;
+}
+
+export interface Columns {
+  field?: string;
+  name: string;
+  isMobileHeader?: boolean;
+  sortable?: boolean;
+  truncateText?: boolean;
+  hideForMobile?: boolean;
+  render?: (item: EventItem) => void;
+}
+
+export class LoadMoreTable extends React.PureComponent<BasicTableProps, BasicTableState> {
+  public readonly state = {
+    isPopoverOpen: false,
+  };
```

Comment:
I don't have an issue with local react state being mixed in, but I was under the impression we wanted all component state to be within redux instead of class state.

No changes requested, just an observation.

Reply:
Local state still makes sense for some components. When a version of `React` that includes hooks is available in Kibana, I would like updates to local state to always be made via the [useReducer](https://reactjs.org/docs/hooks-reference.html#usereducer) API, so we get the benefit of testability (for state updates) that we get with `redux`.

For now, the rule of thumb I've been following is storing non-trivial component state in `redux`, but we may want to refactor some of the code/tests to use `useReducer` when it lands in Kibana.

Until then, you can always code local state updates as pure functions that follow the reducer pattern (`model -> action -> new model`), and then invoke `setState` with the output of said function. 😄